### PR TITLE
Add support for empty field_link in paragraph.hero_image

### PR DESCRIPTION
**Pull request by bmunslow**
#551 

A fatal error was thrown when the optional field `field_link` was left empty in `paragraph.hero_image`.

This commit fixes the issue by conditionally rendering the button in the helper method `ElementTrait::buildElementHeroImage`.

### CODE REVIEW

**Comment by bmunslow** [comment]
image

File: web/modules/custom/server_general/src/ElementTrait.php [hunk 1]
```diff
@@ -69,15 +69,17 @@ protected function buildElementCta(string $title, array $body, string $button_te
...
+   *   If NULL, no button is rendered. Defaults to NULL.
+   * @param \Drupal\Core\Url|null $url
    *   The URL to link the button to.
+   *   If NULL, no button is rendered. Defaults to NULL.
    *
    * @return array
    *   Render array.
    */
-  protected function buildElementHeroImage(array $image, string $title, string $subtitle, string $button_text, Url $url): array {
+  protected function buildElementHeroImage(array $image, string $title, string $subtitle, string $button_text = NULL, Url $url = NULL): array {
```
Comment (amitaibu) [inline web/modules/custom/server_general/src/ElementTrait.php:82]: How about changing this to `Link $link`. This means we won't have weird cases where we have button text, but no button URL.

```suggestion
  protected function buildElementHeroImage(array $image, string $title, string $subtitle, Link $link): array {
```

**Review by amitaibu — COMMENTED** [review]
Thanks! Added a comment 😄

File: web/modules/custom/server_general/src/ElementTrait.php (hunk 1 above)
Comment (bmunslow) [reply web/modules/custom/server_general/src/ElementTrait.php]: Ok, I have amended the commit, please take another look. Notice the `Link` object must have a default `NULL` value.

Also updated the `StyleGuideController` to reflect the change in the parameters passed on to `ElementTrait::buildElementHeroImage`.

**Comment by amitaibu** [comment]
Unfortunately, Travis CI doesn't run on forks, so if you can, please execute manually and attach a screenshot confirming it passes:

- `ddev phpcs` (It should fix a few issues currently in the code)
- `ddev phpstan`
- `ddev phpunit`

**Comment by bmunslow** [comment]
> Unfortunately, Travis CI doesn't run on forks, so if you can, please execute manually and attach a screenshot confirming it passes:
> 
> * `ddev phpcs` (It should fix a few issues currently in the code)
> * `ddev phpstan`
> * `ddev phpunit`

Sure thing!

I amended the commit again with the changes fixed by `ddev phpcs`, it alphabetically ordered `Use` statements, thanks for that.

`ddev phpstan `and `ddev phpunit` also pass cleanly:

image
image
image
image
image
image
image

**Review by amitaibu — APPROVED** [review]
👍  Thanks

**Pull request merged**